src/main/java/org/takes/facets/auth/PsByFlag.java

```diff
@@ -98,19 +97,22 @@ public PsByFlag(final String flg, final Map<Pattern, Pass> map) {
     }
 
     @Override
-    public Iterator<Identity> enter(final Request req) throws IOException {
+    public Opt<Identity> enter(final Request req) throws IOException {
         final Iterator<String> flg = new RqHref.Base(req).href()
             .param(this.flag).iterator();
-        final Collection<Identity> users = new ArrayList<Identity>(1);
+        Opt<Identity> retValue = null;
```

Comment:
Ok, I will rename that, but I don't agree with the ideia of initializing the variable. IMHO, this can be a waste of process and use of memory if this is going to be initialized again.